File: Server/src/main/java/cn/arorms/hsi/server/entities/AuditLog.java
package cn.arorms.hsi.server.entities;

import cn.arorms.hsi.server.enums.LogLevel;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * Entity representing an audit log entry.
 * 
 * @author Cacciatore
 * @version 1.0 2026-03-01
 */
@Data
@Entity
@Table(name = "audit_logs")
public class AuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private LogLevel level;

    private String content;

    @ManyToOne
    @JoinColumn(name = "operator_id")
    private User operator;

    private LocalDateTime timestamp = LocalDateTime.now();
}

File: Server/src/main/kotlin/cn/arorms/hsi/server/entities/AuditLog.kt
package cn.arorms.hsi.server.entities

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "system_logs")
data class AuditLog(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    var timestamp: LocalDateTime = LocalDateTime.now(),

    @ManyToOne
    @JoinColumn(name = "operator_id")
    var operator: User? = null
) {
    constructor(operator: User?) : this() {
        this.timestamp = LocalDateTime.now()
        this.operator = operator
    }
}
